### PR TITLE
docs: adding lets block

### PR DESCRIPTION
**Pull request by AnnuCode**

### CODE REVIEW

File: cli/reference/blocks/lets.md [hunk 1]
```diff
@@ -4,3 +4,79 @@ description: Learn how to define variables available to the current generate_hcl
...
+    greeting = "Hello, World!"
+  }
+
+  content = <<EOF
+  {
+    "message": "${let.greeting}"
+  }
+  EOF
+}
+```
```
Comment (i4ki) [inline cli/reference/blocks/lets.md:82]: `generate_hcl` expects a `content` block.

**Review by i4ki — DISMISSED** [review]

File: cli/reference/blocks/lets.md (hunk 1 above)
Comment (AnnuCode) [reply cli/reference/blocks/lets.md:82]: can you please tell me more about it?

**Review by soerenmartius — APPROVED** [review]
lgtm

File: cli/reference/blocks/lets.md [hunk 2]
```diff
@@ -4,3 +4,81 @@ description: Learn how to define variables available to the current generate_hcl
...
+  }
+
+  content {
+    body = <<EOF
+    {
+      "message": "${let.greeting}"
+    }
+    EOF
+  }
+}
```
Comment (i4ki) [inline cli/reference/blocks/lets.md:83]:
```suggestion
  content {
    message: let.greeting
  }
}
```
you don't need a template string with `generate_hcl`, that's why it exists, so it become easier to do this kind of stuff.

**Review by i4ki — CHANGES_REQUESTED** [review]

**Review by i4ki — APPROVED** [review]
LGTM

**Pull request merged**